packages/hdwallet-native/src/bitcoin.ts

```diff
@@ -193,13 +195,13 @@ export function MixinNativeBTCWallet<TBase extends core.Constructor<NativeHDWall
           case "p2sh-p2wpkh":
           case "p2sh":
           case "bech32":
-            scriptData.redeemScript = payment.redeem?.output;
+            scriptData.redeemScript = payment.redeem?.output ? Buffer.from(payment.redeem.output) : undefined;
```

Comment:
The exact same as before (redeem script or undefined) just honouring `payment.redeem.output`now being an Uint8Array, and `scriptData.redeemScript` now being a `Buffer`, meaning we need a ternary here